test/testvm.py

```diff
@@ -731,23 +749,37 @@ def _cleanup(self):
             print >> sys.stderr, "WARNING: Cleanup failed:", str(value)
 
     def kill(self):
+        # Attempt graceful shutdown first
+        if self._process and self._process.poll() is None:
+            self.execute(command="poweroff", ignore_ret_code=True)
```

Comment:
I had issues during testing where the vm's weren't properly killed and I had zombie processes. This may be outdated, since we've since moved to calling qemu directly, not via shell.